ducktape/cluster/vagrant.py

```diff
@@ -24,10 +28,18 @@ class VagrantCluster(JsonCluster):
     advertised, this assumes that the Vagrant VM's name is a routeable hostname on all the hosts.
     """
 
-    def __init__(self):
+    def __init__(self, *args, **kwargs):
         hostname, ssh_hostname, username, flags = None, None, None, ""
         nodes = []
 
+        if kwargs.get("use_cluster_info", False):
```

Comment:
Is there a reason this isn't in `JsonCluster`? It seems like it's cached json so it's equally applicable.

Also, by moving loading from file exclusively into Vagrant cluster, I think we now can't instantiate a non-Vagrant json cluster from file